src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,128 +1,106 @@
-     * Верните измененное время на указаную величину.
-     */
     public LocalTime addMinutes(LocalTime localTime, Integer minutesToAdd) {
-        return LocalTime.now();
+        return localTime.plusMinutes(minutesToAdd);
     }
 
-    /**
-     * Дано время и на сколько секунд нужно его изменить.
-     * Верните измененное время на указаную величину.
-     */
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
 
-    /**
-     * Дана дата и на сколько недель нужно ее изменить.
-     * Верните получившуюся дату
-     */
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
-    /**
-     * Дана произвольная дата getDate.
-     * Определите соотношение сегодня к someDate и верните строку:
-     * - "someDate is after текущая дата" - если getDate в будующем
-     * - "someDate is before текущая дата" - если getDate в прошлом
-     * - "someDate is today" - если someDate - сегодня
-     */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate localDate = LocalDate.now();
+        return someDate.isBefore(localDate) ? someDate + " is before " + localDate
+                : someDate.isAfter(localDate) ? someDate + " is after " + localDate
+                : someDate + " is today";
     }
 
-    /**
-     * Дана дата в строковом формате и временная зона.
-     * Верните LocalDateTime в этой временной зоне.
-     * @return LocalDateTime
-     */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+
+        return ZonedDateTime.parse(dateInString)
+                .withZoneSameInstant(ZoneId.of(zone))
+                .toLocalDateTime();
     }
 
-    /**
-     * Данны дата и время. Надо вернуть дату и время с местным временным смещением,
-     * пусть это будет для Украины "+02:00".
-     * Приведем пример: при вызове метода передается переменная типа LocalDateTime,
-     * в формате "2019-09-06T13:17", нам надо вернуть переменную типа OffsetDateTime,
-     * в формате "2019-09-06T13:17+02:00", где "+02:00" и будет смещение для нашей
-     * временной зоны.
-     * OffsetDateTime советуют использовать при записи даты в базу данных.
-     */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return OffsetDateTime.of(localTime, ZoneOffset.ofHours(2));
     }
 
-    /**
-     * Дана строка в виде "yyyymmdd".
-     * Необходимо вернуть Optional даты в LocalDate формате
-     */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.parse(date,
+                    DateTimeFormatter.ofPattern("yyyyMMdd")));
```

Comment:
```suggestion
                    DateTimeFormatter.BASIC_ISO_DATE));
```
you can use here formatter that already exists

Reply:
done